### ISSUE TITLE
Wrongly generated Mapper for nested fields

### ISSUE THREAD

**Issue by madness-inc** [mapstruct/mapstruct#2359]
Hi Folks,

I have the following classes (getters and setters omitted for readability):

```java
public class Source {
	private Nested nested;
}

public class Nested {
	private String value1;
	private String value2;
}

public class Target {
	private Wrapper wrapper1;
	private Wrapper wrapper2;
}

public class Wrapper {
	private String value;
}
```

When defining the following Mapper:

```java
@Mapper
public abstract class TargetMapper {

	@Mapping(target = "wrapper1.value", source = "source.nested.value1")
	@Mapping(target = "wrapper2.value", source = "source.nested.value2")
	public abstract Target map(Source source);
}
```

the generated `TargetMapperImpl` maps both target fields `wrapper1.value`and `wrapper2.value` to `source.nested.value1`, which is wrong:

```java
public class TargetMapperImpl extends TargetMapper {

    @Override
    public Target map(Source source) {
        if ( source == null ) {
            return null;
        }

        Target target = new Target();

        target.setWrapper1( nestedToWrapper( source.getNested() ) );
        target.setWrapper2( nestedToWrapper( source.getNested() ) );

        return target;
    }

    protected Wrapper nestedToWrapper(Nested nested) {
        if ( nested == null ) {
            return null;
        }

        Wrapper wrapper = new Wrapper();

        wrapper.setValue( nested.getValue1() );

        return wrapper;
    }
}
```

Is this a bug (I think so!), or am I missing something?

I've created a [small project ](https://github.com/madness-inc/3rd-party-issues/tree/master/mapstruct-issue-nested-fields)to demonstrate this.

Regards
Matthias

**Comment by wangfengqi2010** [comment]
@madness-inc upgrade to lastest version. it's work fine.

**Comment by wangfengqi2010** [comment]
@madness-inc #2236 
![image](https://user-images.githubusercontent.com/10048504/108582960-3ef9f780-7371-11eb-8a97-2bdd7fe654a4.png)

**Comment by madness-inc** [comment]
Hi @wangfengqi2010,

thanks for finding this, you're absolutely right. Commit graph never lies ;-)

Ticket can be closed as a duplicate of #2236

Cheers
Matthias

**Comment by filiphr** [comment]
Thanks @wangfengqi2010 for the clarification.
